mysql-test/lib/My/SafeProcess/safe_process.cc

```diff
@@ -89,60 +105,78 @@ static void die(const char* fmt, ...)
 }
 
 
-static void kill_child(void)
+static void wait_pid(void) 
 {
   int status= 0;
 
-  message("Killing child: %d", child_pid);
-  // Terminate whole process group
-  kill(-child_pid, SIGKILL);
+  pid_t ret_pid;
+  while ((ret_pid = waitpid(child_pid, &status, 0)) < 0) 
```

Comment:
safe_process would be killed by mysql-test-run.pl upon reaching testcase timeout